File: Proj4a/A de Peter e Robinson/PeterRobinson.c
#include <stdio.h>

// Função A de Peter e Robinson
int A(int m, int n) {
    if (m == 0) return n + 1;
    if (n == 0) return A(m-1, 1);
    return A(m-1, A(m, n-1));
}

int main() {

    // Testando a função A de Peter e Robinson
    printf("A(3, 15) = %d\n", A(3, 15));

    return 0;
}

File: Proj4a/A de Peter e Robinson/PeterRobinson.java
public class PeterRobinson {

    // Função A de Peter e Robinson
    public static int A(int m, int n) {
        if (m == 0) return n + 1;
        if (n == 0) return A(m - 1, 1);
        return A(m - 1, A(m, n - 1));
    }

    public static void main(String[] args) {

        // Testando a função A de Peter e Robinson
        System.out.println("A(3, 15) = " + A(3, 12));
    }
}
